site/content/posts/2021-08-27-supporting-ad-oidc-workflows.md

```diff
@@ -0,0 +1,115 @@
+
+In our previous post on LDAP, we mentioned that the reason to support LDAP and AD was primarily to help the cluster administrator easily manage and configure these Identity Providers using Kubernetes APIs. Some of the available identity shims, such as Dex and UAA, can be used between Pinniped and the Identity providers, but they are difficult to configure and the cluster administration may not be able to manage their Day 2 operations using Kubernetes APIs.  
+
+Our initial LDAP implementation released with v.10.0 can be used to work with any LDAP based Identity Provider including Active Directory, but with this release we provide APIs that are specifically tailored to the Active Directory configuration.
+
+### Setup and Use AD with your Supervisor
+
+Pinniped Supervisor authenticates your users with the AD provider via the LDAP protocol, and then issues unique, short-lived, per-cluster tokens. Our previous blog post on [LDAP configuration]({{< ref "2021-06-02-first-ldap-release.md">}}), elaborates on the security considerations to support integration at the Pinniped Supervisor level instead of at the Concierge.
+
+To setup the AD configuration, once you have Supervisor configured with ingress [installed the Pinniped Supervisor]({{< ref "docs/howto/install-supervisor.md" >}}) and you have [configured a FederationDomain]({{< ref "docs/howto/configure-supervisor" >}}) to issue tokens for your downstream clusters, you can create an [ActiveDirectoryIdentityProvider](https://github.com/vmware-tanzu/pinniped/blob/main/generated/1.20/README.adoc#activedirectoryidentityprovider) in the same namespace as the Supervisor.
+Here’s what an example configuration looks like
+
+```yaml
+ apiVersion: idp.supervisor.pinniped.dev/v1alpha1
+ kind: ActiveDirectoryIdentityProvider
+ metadata:
+   name: my-active-directory-idp
+   namespace: pinniped-supervisor
+ spec:
+
+   # Specify the host of the Active Directory server.
+   host: "activedirectory.example.com:636"
+
+   # Specify the name of the Kubernetes Secret that contains your Active Directory
+   # bind account credentials. This service account will be used by the
+   # Supervisor to perform LDAP user and group searches.
+   bind:
+     secretName: "active-directory-bind-account"
+
+ ---
+
+ apiVersion: v1
+ kind: Secret
+ metadata:
+   name: active-directory-bind-account
+   namespace: pinniped-supervisor
+ type: kubernetes.io/basic-auth
+ stringData:
+
+   # The dn (distinguished name) of your Active Directory bind account.
+   username: "CN=Bind User,OU=Users,DC=activedirectory,DC=example,dc=com"
+
+   # The password of your Active Directory bind account.
+   password: "YOUR_PASSWORD"
+ ```
+
+You can also customize the userSearch and groupSearch as shown in the examples in our reference documentation [here] ({{< ref “docs/howto/configure-supervisor-with-activedirectory.md" >}})
```

Comment:
It looks like the double quotes here are not the standard double quote characters, probably as a result of copy/paste from a Google Doc. It causes this error:

`Error: Error building site: "/opt/build/repo/site/content/posts/2021-08-27-supporting-ad-oidc-workflows.md:63:127": unrecognized character in shortcode action: U+201C '“'. Note: Parameters with non-alphanumeric args must be quoted`

Reply:
Resolved in latest changes